### ISSUE TITLE
Count methods for the test apk

### ISSUE THREAD

**Issue by plastiv** [KeepSafe/dexcount-gradle-plugin#230]
In kotlin, rxjava era having single dex only limitation for the test-apk emerges the need to control methods count for the project `androidTestImplementation` dependencies.

**Comment by benjamin-bader** [comment]
I'm not sure that I understand the issue; let me rephrase, and tell me if I got it right:

Test APKs are limited to a single .dex file.  When using large libraries like rxjava and kotlin-stdlib, it's easy to go over the limit.  Therefore, we need to add the ability to count methods in test APKs.

Is that right?  If so, I can't think off-hand how that might be implemented.  Any ideas?

**Comment by ZakTaccardi** [comment]
Libraries like RxJava and Kotlin are a part of your main application, so they don't get bundled into your test app, so they don't count against your test apk's Dex limit. Libs like Mockito or Wiremock are test only, so they do count against that limit.

Test apks cannot multidex and are limited to a single Dex file pre-lollipop.

**Comment by plastiv** [comment]
> Is that right? If so, I can't think off-hand how that might be implemented. Any ideas?

Sry for the confusion. Yes, you are correct. Test apk are builded similar to the main one and there are `assembleAndroidTest` gradle tasks available. Would it be possible to reuse their output for processing?

**Comment by benjamin-bader** [comment]
Sorry to let this sit; I don't actually know how this might be implemented, and don't have the time to research it right now.

I'll leave this open for now, and if someone wants to make an attempt, I'd welcome the help!

**Comment by chaitraraj** [comment]
I had a question related to test apk's.
Currently I have an espresso test apk (3.8Mb) which earlier (2 MB) did not have multiple dex files but after a while started having multiple dex files. I am trying to find out what library might have added to the sudden increase in size and caused multiple dex files.I did use the APK analyzer in Android studio but it did not help .

**Comment by benjamin-bader** [comment]
Your question is unrelated to this issue - stackoverflow is a more appropriate venue for this kind of question.  If you have a question about the dexcount plugin specifically, please open a new issue.

...but in your case, where method counts have gone up suddenly, source control is your friend.  Look at git logs and see what changed.  Maybe `git bisect` if you're totally unsure.

**Comment by autonomousapps** [comment]
I would be in favor of adding support for counting the methods of a test APK. This has bitten us a couple times. I recently tried this:
```
public final class MultiDexTestRunner extends AndroidJUnitRunner {

    @Override
    public void onCreate(Bundle arguments) {
        MultiDex.install(getTargetContext());
        super.onCreate(arguments);
    }
}
```
and actually I'm not sure it works! But it doesn't seem to _break_ anything.

**Comment by benjamin-bader** [comment]
Given that I don't have much discretionary time to spend, I'm going to close this as won't-fix.  It sounds like this may be a) a pre-Lollipop issue, and b) work-aroundable.
